code/game/objects/items/weapons/pneumaticCannon.dm

```diff
@@ -29,57 +37,61 @@
+	if(I.flags & (ABSTRACT | NODROP | DROPDEL))
+		to_chat(user, "<span class='warning'>You can't put [I] into [src]!</span>")
+		return FALSE
+	if((loaded_weight_class + I.w_class) > max_weight_class)
+		to_chat(user, "<span class='warning'>\The [I] won't fit into \the [src]!</span>")
+		return FALSE
+	if(I.w_class > src.w_class)
+		to_chat(user, "<span class='warning'>\The [I] is too large to fit into \the [src]!</span>")
+		return FALSE
+	if(!user.unEquip(I))
+		return FALSE
+	to_chat(user, "<span class='notice'>You load \the [I] into \the [src].</span>")
+	loaded_items.Add(I)
+	loaded_weight_class += I.w_class
+	I.forceMove(src)
+	return TRUE
+
+/obj/item/pneumatic_cannon/wrench_act(mob/living/user, obj/item/I)
+	switch(pressure_setting)
+		if(1)
+			pressure_setting = 2
+		if(2)
+			pressure_setting = 3
+		if(3)
+			pressure_setting = 1
+	to_chat(user, "<span class='notice'>You tweak \the [src]'s pressure output to [pressure_setting].</span>")
+	return TRUE
+
+
 /obj/item/pneumatic_cannon/attackby(obj/item/W, mob/user, params)
 	..()
-	if(istype(W, /obj/item/tank/internals/) && !tank)
+	if(istype(W, /obj/item/tank/internals) && !tank)
 		if(istype(W, /obj/item/tank/internals/emergency_oxygen))
 			to_chat(user, "<span class='warning'>\The [W] is too small for \the [src].</span>")
 			return
-		updateTank(W, 0, user)
+		add_tank(W, user)
 		return
 	if(W.type == type)
 		to_chat(user, "<span class='warning'>You're fairly certain that putting a pneumatic cannon inside another pneumatic cannon would cause a spacetime disruption.</span>")
 		return
-	if(istype(W, /obj/item/wrench))
-		switch(pressureSetting)
-			if(1)
-				pressureSetting = 2
-			if(2)
-				pressureSetting = 3
-			if(3)
-				pressureSetting = 1
-		to_chat(user, "<span class='notice'>You tweak \the [src]'s pressure output to [pressureSetting].</span>")
-		return
-	if(loadedWeightClass >= maxWeightClass)
-		to_chat(user, "<span class='warning'>\The [src] can't hold any more items!</span>")
-		return
-	if(isitem(W))
-		var/obj/item/IW = W
-		if(IW.flags & (ABSTRACT | NODROP | DROPDEL))
-			to_chat(user, "<span class='warning'>You can't put [IW] into [src]!</span>")
-			return
-		if((loadedWeightClass + IW.w_class) > maxWeightClass)
-			to_chat(user, "<span class='warning'>\The [IW] won't fit into \the [src]!</span>")
-			return
-		if(IW.w_class > src.w_class)
-			to_chat(user, "<span class='warning'>\The [IW] is too large to fit into \the [src]!</span>")
-			return
-		if(!user.unEquip(W))
-			return
-		to_chat(user, "<span class='notice'>You load \the [IW] into \the [src].</span>")
-		loadedItems.Add(IW)
-		loadedWeightClass += IW.w_class
-		IW.loc = src
-		return
+	load_item(W, user)
 
 /obj/item/pneumatic_cannon/screwdriver_act(mob/living/user, obj/item/I)
-	if(!tank)
-		return
-
-	updateTank(tank, 1, user)
+	remove_tank(user)
 	return TRUE
 
 /obj/item/pneumatic_cannon/afterattack(atom/target, mob/living/carbon/human/user, flag, params)
```

Comment:
I'm unsure about this part or how much it's actually needed. I'll test it more tomorrow.